dace/codegen/targets/mlir/utils.py

```diff
@@ -1,97 +1,124 @@
 '''
 Helps extracting the relevant information from MLIR for CodeGen of an MLIR tasklet
+Can handle MLIR in generic form or in the supported dialect of pyMLIR
+Requires pyMLIR to run
 '''
 import dace
 
-class MLIRUtils():
-
-    TYPE_DICT = {
-        "i8": dace.int8,
-        "i16": dace.int16,
-        "i32": dace.int32,
-        "i64": dace.int64,
-        "f16": dace.float16,
-        "f32": dace.float32,
-        "f64": dace.float64
-    }
-
-    def __init__(self, code, func_uid=""):
-        try:
-            self.mlir =  __import__('mlir')
-        except (ModuleNotFoundError, NameError, ImportError):
-            raise ImportError('To use MLIR tasklets, please install the "pymlir" package.')
-
-        self.code = code
-        self.entry_func_name = "mlir_entry" + func_uid
-
-        self.ast = self.mlir.parse_string(self.code)
-        self.is_generic = isinstance(self.ast, self.mlir.astnodes.GenericModule)
-
-        # Iterating over every function in the body to check if ast contains exactly one entry and saving the entry function
-        self.entry_func = None
-        for func in self.ast.body: 
-            func_name = self.__get_func_name(func)
-
-            if func_name == self.entry_func_name:
-                if self.entry_func is not None:
-                    raise SyntaxError("Multiple entry function in MLIR tasklet.")
-                self.entry_func = func
-
-        if self.entry_func is None:
-            raise SyntaxError('No entry function in MLIR tasklet, please make sure a "mlir_entry()" function is present.')
-
-    def __get_func_name(self, func):
-        if self.is_generic:
-            # In generic ast the name can be found in ast->body->func[]->attributes->values[0]->value
-            # The consecutive .values ensure to read the name as a string
-            return func.attributes.values[0].value.value.value 
-        # In dialect ast the name can be found in ast->body->func[]->name->value
-        return func.name.value
-        
-    def get_entry_args(self):
-        ret = []
-
-        if self.is_generic:
-            # In generic ast the list of arguments can be found in ast->body->func[]->body[0]->label->arg_ids
-            arg_names = self.entry_func.body[0].label.arg_ids
-            arg_types = self.entry_func.body[0].label.arg_types
-
-            for idx in range( len(arg_names) ):
-                arg_name = arg_names[idx].value
-                arg_type = arg_types[idx]
-                ret.append( (arg_name, arg_type) )
-            return ret
-
-        for arg in self.entry_func.args:
-            arg_name = arg.name.value
-            arg_type = arg.type
-            ret.append( (arg_name, arg_type) )
+# Only these types and the vector version of them are supported
+TYPE_DICT = {
+    "i8": dace.int8,
+    "i16": dace.int16,
+    "i32": dace.int32,
+    "i64": dace.int64,
+    "f16": dace.float16,
+    "f32": dace.float32,
+    "f64": dace.float64
+}
+
+
+def get_pymlir():
```

Comment:
couldn't you just import mlir at the top level code in this file? If it doesn't get imported anyway... Also, you can catch the exception there.